### PR TITLE
Issue 478 - Add GraphQL feature to CustomFieldChoice

### PR DESCRIPTION
**Pull request by chadell**
### Fixes: #478 

Extends CustomFieldChoice with GraphQL extra feature.

image

image

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
Can you show how a graphql query for `custom_field_choices` returns now?

**Comment by chadell** [comment]
> Can you show how a graphql query for `custom_field_choices` returns now?

are the images attached now descriptive enough?

**Comment by glennmatthews** [comment]
Thanks! Can you filter them  in GraphQL by associated custom field, a la `custom_field_choices(custom_field="cf_test_2")` or is that something that needs to be additionally implemented?

**Comment by chadell** [comment]
> Thanks! Can you filter them in GraphQL by associated custom field, a la `custom_field_choices(custom_field="cf_test_2")` or is that something that needs to be additionally implemented?

It was not supported, but I added this on 20f5a9d
image

File: nautobot/extras/filters.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ class CustomFieldChoiceFilterSet(BaseFilterSet):
 
     class Meta:
         model = CustomFieldChoice
-        fields = ["id", "value", "weight"]
+        fields = ["id", "value", "weight", "field"]
```
Comment (glennmatthews) [inline nautobot/extras/filters.py:154]: Hmmm. Usually with django_filters it's not necessary or appropriate to include explicitly declared Filters (`field = django_filters.ModelMultipleChoiceFilter`, above) in `fields`:

https://django-filter.readthedocs.io/en/stable/ref/filterset.html#declaring-filterable-fields

> Note that it is *not* necessary to included declared filters in a `fields` list - doing so will have no effect

I'll try and find some time today to play around with this and see if I can figure out why `field` isn't automatically being made available as a filter.

Comment (glennmatthews) [reply nautobot/extras/filters.py:154]: Works for me without this change:

image

image

Comment (chadell) [reply nautobot/extras/filters.py]: your are right 😓 , don't why I got the other conclusion

**Review by glennmatthews — APPROVED** [review]
Thank you!

**Pull request merged**